### PR TITLE
clean cache bewteen different TestCafe runs (attempt to fix #5432)

### LINKED ISSUE #5432: Require.cache should be clean for a new TestCafe instance
[BASED ON https://github.com/DevExpress/testcafe-studio/issues/3133]
Given we have a test file `example.js`
```js
import { say } from './test-lib';

fixture('Stale JS')
    .page('about:blank');

test('Function from external file', async t => {
    await say();
    await t
        .debug()
        .expect(true).ok();
});
```
`test-lib.js`
```js
import { ClientFunction } from 'testcafe';

export async function say() {
    await ClientFunction(() => { document.write('FOO'); })();
}
```
and we try to create two TestCafe instances to run this test. **This is similar to how we run tests in TestCafe Studio.**
```js
const createTestCafe = require('testcafe');
let testcafe         = null;

createTestCafe()
    .then(tc => {
        testcafe     = tc;
        const runner = testcafe.createRunner();

       // NOTE: make file changes in test-lib before the timeout is expired
      // FOO -> BAR
        setTimeout(function () {
            createTestCafe()
                .then(tc2 => {
                    const runner2 = tc2.createRunner();

                    runner2
                        .src('test.js')
                        .browsers(['chrome'])
                        .run();
                })
        }, 30000);

        return runner
            .src('test.js')
            .browsers(['chrome'])
            .run();
    });
```
**PROBLEM:** We get the same result on the testing page for both TestCafe instances. 
But if two instances of TestCafe are created in different processes, changes made in the `test-lib` file will be applied properly.
They also will be applied properly if we create two instances of TestCafe in the same process, but the `example.js` will have the ClientFunction instead of using `test-lib` dependency.

It seems like the problem is in the require cache. The initial example works if we will clean `test-lib` module in require cache before the second TestCafe instance creation.
```js
delete require.cache['C:\\GitHub\\testcafe-studio\\test-lib.js'];
```

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

**Review by AndreyBelym — APPROVED** [review]
The failed test actually cannot pass in `master` when started alone. I think we can fix it outiside of this PR.

File: src/compiler/test-file/api-based.js [hunk 1]
```diff
@@ -41,13 +44,22 @@ export default class APIBasedTestFileCompilerBase extends TestFileCompilerBase {
             .includes(NODE_MODULES);
     }
 
-    static _execAsModule (code, filename) {
+    static _isTestCafeLibDep (filename) {
+        return relative(CWD, filename)
+            .split(pathSep)[0] === 'lib';
```
Comment (helen-dikareva) [inline src/compiler/test-file/api-based.js:49]: (minor)
maybe it's better to extract `'lib'` as s const

**Review by helen-dikareva — APPROVED** [review]
It works properly for the initial case in Studio

**Comment by AlexKamaev** [comment]
Do not merge this PR please. It seems that the tests fail consistently.

**Comment by AlexKamaev** [comment]
I removed special cache key for the first compiler.
I disabled this new approach for live mode

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by miherlosev — APPROVED** [review]

**Review by AndreyBelym — APPROVED** [review]

**Pull request merged**